### PR TITLE
Test DBInterface / Tables Migration

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
Exploring whether Tables.jl and DBInterface.jl support is worth implementing.

### CODE REVIEW

**Comment by jeremiahpslewis** [comment]
@kimmolinna Hey! What do you think about implementing Tables.jl and DBInterface.jl and making this a bit of an SQLite clone?

**Comment by jeremiahpslewis** [comment]
@quinnj Quick question...the SQLite implementation is beautiful and can be used here 1:1 at times...what is the best way to ensure there are no license/copyright issues / you get due credit for the copy-paste?

**Comment by kimmolinna** [comment]
I think It would be wise to implement Tables.jl and DBInterface.jl. But you should remember that those use a rowwise approach which is a bit slower one than a columnwise.

**Comment by kimmolinna** [comment]
Looks good.

**Review by kimmolinna — APPROVED** [review]

**Comment by kimmolinna** [comment]
I don't have any probem with credits. I just wanted to help people to use DuckDB database which is a great tool.

File: Project.toml [hunk 1]
```diff
@@ -3,11 +3,13 @@ uuid = "d2f5444f-75bc-4fdf-ac35-56f514c445e1"
...
 [deps]
+DBInterface = "a10d1c49-ce27-4219-8d33-6db1a4562965"
 DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 DuckDB_jll = "2cbbab25-fc8b-58cf-88d4-687a02676033"
+Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
+
+[compat]
+DataFrames = "1.2"
+julia = "1.6"
```
Comment (quinnj) [inline Project.toml:15]: For package registration, you'll want to add compat for DBInterface and Tables

File: src/DuckDB.jl [hunk 2]
```diff
@@ -118,13 +132,92 @@ Note that after running duckdb_query, duckdb_destroy_result must be called on th
...
+struct DuckDBException <: Exception
+    msg::AbstractString
+end
+
+duckdb_errmsg(handle) = unsafe_string(handle[].error_message)
+duckdbexception(handle::DBHandle) = DuckDBException(duckdb_errmsg(handle))
+duckdberror(handle::DBHandle) = throw(duckdbexception(handle))
+
+"""
+    `DuckDB.DB()` => in-memory SQLite database
```
Comment (quinnj) [inline src/DuckDB.jl:153]: in memory `DuckDB` database?

**Comment by quinnj** [comment]
@jeremiahpslewis, thanks for the ping! I'm glad the SQLite.jl code is helpful here. I don't have any issues with it being used here or that it needs to be documented/credited or anything. The beauty of MIT license. Excited to see more database support in Julia!

**Pull request merged**